arch/arm/src/common/arm_createstack.c

```diff
@@ -102,26 +102,26 @@
 
 int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 {
-  /* Add the size of the TLS information structure */
+  size_t alloc_size;
 
-  stack_size += sizeof(struct tls_info_s);
+  /* Add the size of the TLS information structure and align */
+
+  alloc_size = STACK_ALIGN_UP(stack_size + sizeof(struct tls_info_s));
 
 #ifdef CONFIG_TLS_ALIGNED
   /* The allocated stack size must not exceed the maximum possible for the
    * TLS feature.
    */
 
-  DEBUGASSERT(stack_size <= TLS_MAXSTACK);
-  if (stack_size >= TLS_MAXSTACK)
+  DEBUGASSERT(alloc_size <= TLS_MAXSTACK);
```

Comment:
Isn't this assertion backward?
```suggestion
  DEBUGASSERT(alloc_size > TLS_MAXSTACK);
```

We will take the assertion if the condition is false.